src/gwt/src/org/rstudio/studio/client/projects/ui/prefs/ProjectPreferencesDialog.java

```diff
@@ -153,16 +157,74 @@ public void onError(ServerError error)
       
    }
    
-   private void emitRenvConsoleActions(RProjectRenvOptions options)
+   private void emitRenvConsoleActions(RProjectRenvOptions newOptions)
    {
-      if (options.useRenv == renvOptions_.useRenv)
+      EventBus events = pEventBus_.get();
+      
+      // renv is off and staying off -- nothing to do
+      if (!newOptions.useRenv && !renvOptions_.useRenv)
+      {
+      }
+      
+      // turning renv off -- just call deactivate
+      else if (!newOptions.useRenv && renvOptions_.useRenv)
+      {
+         String action = "renv::deactivate()";
+         events.fireEvent(new SendToConsoleEvent(action, true, true));
          return;
+      }
       
-      String renvAction = options.useRenv
-            ? "renv::activate()"
-            : "renv::deactivate()";
+      // turning renv on -- just call activate
+      //
+      // TODO: we need to apply project options here, but we need to call
```

Comment:
We generally avoid committing TODOs to master -- can this be tracked in an issue? (Or can we describe what exactly will be broken before the work is completed?)